src/main/webapp/resources/js/pages/search/SearchLayout.tsx

```diff
@@ -0,0 +1,312 @@
+    content: SearchProject[];
+    total: number;
+  }>();
+  const [projectsTableParams, setProjectsTableParams] = useState<TableParams>(
+    JSON.parse(initial_table_params)
+  );
+  const debouncedSetProjectTableParams = debounce(async (params) => {
+    setProjectsTableParams(params);
+  }, 300);
+
+  const [samples, setSamples] = useState<{
+    content: SearchSample[];
+    total: number;
+  }>();
+  const [samplesTableParams, setSamplesTableParams] = useState<TableParams>(
+    JSON.parse(initial_table_params)
+  );
+  const debouncedSetSamplesTableParams = debounce(async (params) => {
+    setSamplesTableParams(params);
+  }, 300);
+
+  const handleProjectsTableChange: TableProps<SearchProject>["onChange"] = (
+    pagination,
+    filters,
+    sorter
+  ): void => {
+    debouncedSetProjectTableParams({
+      pagination,
+      filters,
+      ...sorter,
+    });
+  };
+
+  const handleSamplesTableChange: TableProps<SearchSample>["onChange"] = (
+    pagination,
+    filters,
+    sorter
+  ): void => {
+    debouncedSetSamplesTableParams({
+      pagination,
+      filters,
+      ...sorter,
+    });
+  };
+
+  const fetchData = useCallback(() => {
+    const fetchSamples = async () =>
+      axios.post(setBaseUrl(`/ajax/search/samples`), {
+        global,
+        pagination: samplesTableParams.pagination,
+        order: [
+          {
+            property: samplesTableParams.columnKey || `sampleName`,
+            direction: samplesTableParams.order === "ascend" ? `asc` : `desc`,
+          },
+        ],
+        search: [
+          {
+            property: `name`,
+            value: searchParams.get("query"),
+            operation: "MATCH_IN",
+          },
+        ],
+      });
+
+    const fetchProjects = async () =>
+      axios.post(setBaseUrl(`/ajax/search/projects`), {
+        global,
+        pagination: projectsTableParams.pagination,
+        order: [
+          {
+            property: projectsTableParams.columnKey || `name`,
+            direction: projectsTableParams.order === "ascend" ? `asc` : `desc`,
+          },
+        ],
+        search: [
+          {
+            property: `name`,
+            value: searchParams.get("query"),
+            operation: "MATCH_IN",
+          },
+        ],
+      });
+
+    const promises = [];
+    promises.push(fetchProjects());
+    promises.push(fetchSamples());
+    Promise.all(promises).then(
+      ([{ data: projectsData }, { data: samplesData }]) => {
+        setProjects(projectsData);
+        setSamples(samplesData);
+      }
+    );
+  }, [
+    global,
+    projectsTableParams.columnKey,
+    projectsTableParams.order,
+    projectsTableParams.pagination,
+    samplesTableParams.columnKey,
+    samplesTableParams.order,
+    samplesTableParams.pagination,
+    searchParams,
+  ]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData, global, searchParams]);
+
+  const menuItems = useMemo(
+    () => [
+      {
+        label: (
+          <div
```

Comment:
Can we create a styled component for this div? since it's the same for both menu items

Reply:
Sure, updated in 22954f2